lib/media/gap_jumping_controller.js

```diff
@@ -304,15 +304,25 @@ shaka.media.GapJumpingController = class {
     const threshold = this.config_.stallThreshold;
     const skip = this.config_.stallSkip;
 
-    const onStall = (at, duration) => {
+    const onStall = async (at, duration) => {
       goog.asserts.assert(this.video_, 'Must have video');
+      const bufferedInfo =
+          shaka.media.TimeRangesUtils.getBufferedInfo(this.video_.buffered);
+      if (!bufferedInfo.length) {
+        // Nothing in the buffer.
+        return;
+      }
       shaka.log.debug(`Stall detected at ${at} for ${duration} seconds.`);
 
       if (skip) {
         shaka.log.debug(`Seeking forward ${skip} seconds to break stall.`);
         this.video_.currentTime += skip;
       } else {
         shaka.log.debug('Pausing and unpausing to break stall.');
+        await this.video_.play();
+        if (!this.video_) {
+          return;
+        }
```

Comment:
Log says pausing and unpausing, but first we're hitting play, so looks like a bug at a first look. Could you add a comment with explanation what is going on?

Reply:
Done!